internal/app/machined/pkg/controllers/k8s/kubelet_config.go

```diff
@@ -73,36 +91,49 @@ func (ctrl *KubeletConfigController) Run(ctx context.Context, r controller.Runti
 		if err = r.Modify(
 			ctx,
 			k8s.NewKubeletConfig(k8s.NamespaceName, k8s.KubeletID),
-			func(r resource.Resource) error {
-				kubeletConfig := r.(*k8s.KubeletConfig).TypedSpec()
+			modifyKubeletConfig(cfgProvider, staticPodListURL),
+		); err != nil {
+			return fmt.Errorf("error modifying KubeletConfig resource: %w", err)
+		}
+	}
+}
 
-				kubeletConfig.Image = cfgProvider.Machine().Kubelet().Image()
+func getStaticPodListURL(ctx context.Context, r controller.Runtime) (string, error) {
+	staticPodURLRes, err := r.Get(ctx, resource.NewMetadata(k8s.NamespaceName, k8s.StaticPodServerStatusType, k8s.StaticPodServerStatusResourceID, resource.VersionUndefined))
```

Comment:
```suggestion
	staticPodURLRes, err := safe.ReaderGet[*k8s.StaticPodServerStatus](ctx, r, resource.NewMetadata(k8s.NamespaceName, k8s.StaticPodServerStatusType, k8s.StaticPodServerStatusResourceID, resource.VersionUndefined))
```

use type-safe version to avoid typecasting below